components/movementsensor/gpsnmea/serial.go

```diff
@@ -95,33 +91,25 @@ func (g *SerialNMEAMovementSensor) Start(ctx context.Context) error {
 	g.activeBackgroundWorkers.Add(1)
 	utils.PanicCapturingGo(func() {
 		defer g.activeBackgroundWorkers.Done()
-		r := bufio.NewReader(g.dev)
 		for {
 			select {
 			case <-g.cancelCtx.Done():
 				return
-			default:
+			case line := <-g.dev.Lines():
```

Comment:
I wonder if the compiler is smart enough to just call `Lines()` once, or if it calls it each time through the loop. Maybe it should be moved into a variable before the loop starts, so there's no doubt.